src/Symfony/Bridge/Monolog/Messenger/ResetLoggersWorkerSubscriber.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Monolog\Messenger;
+
+use Monolog\ResettableInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\WorkerMessageFailedEvent;
+use Symfony\Component\Messenger\Event\WorkerMessageHandledEvent;
+
+/**
+ * Reset loggers between messages being handled to release buffered handler logs.
+ *
+ * @author Laurent VOULLEMIER <laurent.voullemier@gmail.com>
+ */
+class ResetLoggersWorkerSubscriber implements EventSubscriberInterface
+{
+    private $loggers;
+
+    public function __construct(iterable $loggers)
+    {
+        $this->loggers = $loggers;
+    }
+
+    public function onWorkerMessageHandled(): void
+    {
+        $this->resetLoggers();
+    }
+
+    public function onWorkerMessageFailed(): void
+    {
+        $this->resetLoggers();
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            WorkerMessageHandledEvent::class => 'onWorkerMessageHandled',
```

Comment:
`onWorkerMessageHandled` and `onWorkerMessageFailed` do nothing and can be removed here.

```suggestion
            WorkerMessageHandledEvent::class => 'resetLoggers',
```

Reply:
Of course, my bad. Thank you